src/main/java/net/minestom/server/utils/NamespaceID.java

```diff
@@ -49,9 +49,15 @@ private NamespaceID(String full, String domain, String path) {
         this.full = full;
         this.domain = domain;
         this.path = path;
-        assert !domain.contains(".") && !domain.contains("/") : "Domain cannot contain a dot nor a slash character (" + full + ")";
-        assert domain.matches(legalLetters) : "Illegal character in domain (" + full + "). Must match " + legalLetters;
-        assert path.matches(legalPathLetters) : "Illegal character in path (" + full + "). Must match " + legalPathLetters;
+        if(domain.contains(".") || domain.contains("/")) {
+            throw new IllegalArgumentException("Domain cannot contain a dot nor a slash character (" + full + ")");
+        }
+        if(!domain.matches(legalLetters)) {
```

Comment:
regex checks are fairly expensive, it is the original reason why these were assertions

Reply:
True, but simply never checking means you can register illegal names.

I could try to split it up into different checks and avoid using regex, would that be what's needed?